### ISSUE TITLE
darshan-merge

### ISSUE THREAD

**Issue by shanedsnyder** [darshan-hpc/darshan#205]
In GitLab by @glennklockwood on Sep 15, 2016, 17:42

I am starting to mess with mmap'ed partial logs that I've generated on Edison using a version of IOR that just skips `MPI_Finalize()` along with with darshan/e7f180d.  Upon trying to merge the partials though, I got the following error:

```
$ darshan-merge --job-end-time "$(date +%s)" --output ahh.darshan ~/ssio/darshan-mmap/nid0015[89]/*.darshan
Error: unable to read BG/Q module record from input log file /global/homes/g/glock/ssio/darshan-mmap/nid00158/glock_ior_id2068043_mmap-log-15272256825854712713-0.darshan.
```

I did not compile with the BG/Q module, and darshan-parser on these files indicates that there's no trace of BG/Q in there.

I've stuffed the partial logs that are throwing this error here:

    /global/project/projectdirs/m888/glock/edison.2068043/

along with exactly what I did in `README.md`.

**Comment by shanedsnyder** [comment]
In GitLab by @shanedsnyder on Sep 20, 2016, 12:05

This error should be resolved as of commits f21728f3  & ab32cf66.

There was a bug causing utilities to still try to grab data from modules that don't have any data in the log file, and some of the module-specific logutils didn't have proper safety checks to avoid this.

I'll also mention that I was additionally getting a segfault when running the merge utility with the `--shared-redux` flag (i.e., the flag that causes merge to reduce globally shared records into a single aggregate record) because the Lustre logutils didn't implement this function. I added a simple implementation that just initializes the aggregated record from the first rank it sees that references that record, then just asserts that all following ranks that have that record have identical data in the counters.

**Comment by shanedsnyder** [comment]
In GitLab by @shanedsnyder on Sep 20, 2016, 12:08

I'll mark this as closed for now, but I'm going to keep testing to make sure the `darshan-merge` and `darshan-diff` utilities are working properly for the new modules before completing the upcoming release.
